### DISCUSSION TITLE
New Trigger Redirect: StageBGID(ID)

### DISCUSSION THREAD

**Discussion by MarioManX1983** [ikemen-engine/Ikemen-GO#2081] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

Currently, all trigger redirects only redirect to other players or their helpers. None of them redirect to background elements of a stage, which can be a major problem if you want a state to activate according to a condition of a stage element.

For example, let's say you want to use an attachedchar to control a background elements movement. As in make it start moving when a trigger condition is met and then move in another direction or stop moving when another trigger condition is met. Sure you could fairly easily do this already by having the element as an animation instead and then coding the necessary states in the attachedchar but wouldn't be easier and help save on coding if we could just use a single state that can directly interact with the element itself?

### Describe the solution you'd like

In order to do just this, I would like to request a new trigger redirect.

StageBGID(ID)

Redirects to the stage background element with the ID number of (ID). Used to detect changes with the elements default/base conditions.

Just one example.

[State -2, Stop Moving Up]
type = ModifyBGCtrl
trigger1 = StageBGID(1), Pos Y = -300
x = 0
y = 0
id = 2

Stops the background element with ID = 1 from moving up when it reaches the Y position of -300 by modifying the background controller that controls its movement. (Basically, the StageBGID(1), Pos Y = -300 part detects the Y position of the background element with the ID of 1 and when the position that is specified is reached, the trigger is activated.)

### Search terms

trigger, triggers, trigger's, redirect, redirects, redirect's

**Comment by potsmugen** [comment]
Redirects really are meant to be characters. If they weren't, you'd need entirely different sets of triggers for each redirection type. What you're suggesting could be done with a single trigger though.

We've been using "var" a lot for such triggers, so that could be something like `StageBGVar(1, Pos Y)`

**Reply by MarioManX1983** [reply]
I just looked over the StageVar() Trigger and the ModifyStageVar() State Controller and I can see how your suggestion could work. However, I thought of something else. How about instead of a new StageBGVar() Trigger, we instead just have new Optional Parameters for the ModifyStageVar() State Controller that are then readable by the StageVar() Trigger?

**Examples**
------------
stagebg.spriteno = (int), (int)
stagebg.actionno = (int)
stagebg.start.x = (int)
stagebg.start.y = (int)
stagebg.alpha.source = (int)
stagebg.alpha.dest = (int)
stagebg.mask = (int)
stagebg.velocity.x = (float)
stagebg.velocity.y = (float)
stagebg.tile.x = (int)
stagebg.tile.y = (int)
stagebg.delta.x = (int)
stagebg.delta.y = (int)
stagebg.layerno = (int)
stagebg.tilespacing.x = (int)
stagebg.tilespacing.y = (int)
stagebg.xscale.top = (float)
stagebg.xscale.bottom = (float)
stagebg.width.top = (int)
stagebg.width.bottom = (int)
stagebg.yscalestart = (float)
stagebg.yscaledelta = (float)
stagebg.id = (int)

Sorry there's a lot there but I wanted to cover everything.

**Reply by potsmugen** [reply]
You'd need to feed in the ID of the BG, which that syntax doesn't allow.

**Reply by LasombraDemon** [reply, marked as answer]
All this is done via BGCtrls and ModifyBGCtrl. :)

If you want help regarding AttachedChars, I think I can lend a hand.
